### PR TITLE
feature: add logging through a service

### PR DESCRIPTION
**Pull request by yamii**

### CODE REVIEW

**Comment by openam** [comment]
I don't know that we should be having lapin send any logging information over rabbitmq. At a minimum lapin shouldn't force it to happen.

**Comment by yamii** [comment]
@openam yeah but we need thorough logs and the best place we can make changes is here in lapin - where we can add logs from request and respond. but i'm open for suggestions

File: lib/req-res.js [hunk 1]
```diff
@@ -36,16 +63,32 @@ module.exports = function ReqRes ( Rabbit ) {
 			}, null );
 		}
 
+		var listener = onErrorEmit( this, log, callback ); // Listen to error emitted
+
+		log.info( {
```
Comment (yamii) [inline lib/req-res.js]: log object is always created regardless if there was logger attached or not. Log object will morph into a `fake` logger when no logger is attached

**Comment by fakewaffle** [comment]
Can we update the README.md with instructions on how to use and turn on/off the logger?

**Comment by yamii** [comment]
@fakewaffle @openam updated README.md

**Pull request merged**